### ISSUE TITLE
Cannot open secure lockers in a custom wreck

### ISSUE THREAD

**Issue by rav2n** [Regalis11/Barotrauma#9156]
These bugs have already been reported, but they continue to appear.

[The problem with the inability to access safes that require a card with a custom ID ](https://github.com/Regalis11/Barotrauma/issues/8098)
and
[Problem with auto-undocking at the start of a round](https://github.com/Regalis11/Barotrauma/issues/6127)

I would like to note that these bugs are visible directly inside the game. And they are not tracked in any way in the subeditor or leveleditor.
Below will be attached a mod-assembly with wreck on which you can meet these problems.
[Test Wrecks.zip](https://github.com/Regalis11/Barotrauma/files/8708753/Test.Wrecks.zip)

**Steps To Reproduce**
1. Start a single campaign or run a mission in multiplayer.
2. Visit the wreks provided in the attached mod.
3. Try to open the safe with the cards taken from the corpses or found on the wreck.
4. ZU-B_Wrecked and Proteus_Wrecked have a undockable bow. The bow must be undocked when the compartments are flooded.

**Version**
0.17.15

**Additional information**
Slightly off topic: if we talk about moving and undocking parts, then their spawn points are not used when generating a level. It is also useless to place the thalamus on a docked object. Perhaps this is too broad a topic, but I could not help touching on it, because the use of such things can be found both for sunken ships and for ruins. I think it’s not worth talking about the wet dreams of communities about elevators at all)

**Comment by Regalis11** [comment]
I'm afraid I could not reproduce this. 

The secure cabinet seems to open with an ID card from the corpses:
![image](https://user-images.githubusercontent.com/14351070/168840034-414b62f6-bc68-4e46-bf6f-94fa18982338.png)

And the docked submarine seems to stay docked:
![image](https://user-images.githubusercontent.com/14351070/168840253-aec05fd1-5048-494a-b5db-0d38ba2caa92.png)

**Comment by rav2n** [comment]
Attached single player save
![image](https://user-images.githubusercontent.com/29980468/168923866-840bac5d-90e8-4f1d-9108-156a77546bd6.png)
[Barotrauma.zip](https://github.com/Regalis11/Barotrauma/files/8712069/Barotrauma.zip)

A bug with access to the safe occurs. But for some reason selectively, it looks like this:
1) SeparatistBase_Wrecked uses ID-card tags: armory_id, secret_id — access is feasible.
2) Nerka_Wrecked1 : wreck_security — no acess.
3) Dingo_Wrecked : wreck_cap — no acess.
4) R-03_Wrecked : wreck_cap, wreck_id, wreck_sec, wreck_med — no acess.  In this case, each access point has two identifiers attached, one of which is wreck_id (wreck_cap,wreck_id)

I couldn’t reproduce the auto-undock bug, there was only a warning in the console about the location of the docking port far from the hull. Although 2 people informed me about auto-undocking.
![image](https://user-images.githubusercontent.com/29980468/168926196-b5488b9c-84d1-4aa6-9a4f-186299c241d6.png)

**Comment by Regalis11** [comment]
Sorry for the extremely late reply. Finally got around to giving this another test while diagnosing [another related issue](https://github.com/Regalis11/Barotrauma/issues/10078), but unfortunately I still couldn't find any issues in the way the secure lockers work in these custom submarines. Every secure container I found seemed to be accessible with the correct card (i.e. the card of the corpse spawning at a spawnpoint that gives the card the tags for opening the container). 

Here's one of the example, Dingo, in which the captain spawning in the command room has a card that opens the secure cabinet in the room below:
![image](https://user-images.githubusercontent.com/14351070/196046997-651efa6a-5d01-477b-94de-92bbeeb01218.png)

Same in R-03:
![image](https://user-images.githubusercontent.com/14351070/196047341-36ea5fd0-5bcf-4b69-9f15-a219dd03c145.png)

Closing as non-reproducible.
